solo-io/gloo
# Fix failure dump bugs

# Description

- Use the passed `path` to `recordEnvoyAdminData()` as an argument to `GetEnvoyAdminData()` instead of hard-coding one particular path
- Add any error returned when retrieving logs from pods to a multi-error and return after all pods have been dumped rather than always short-circuit the `for` loop for logging pods on the first iteration

# Context

We observed that the output in each file of the `envoy-dump` directory in failed test artifacts was the same despite claiming to come from different endpoints

We also observed that since #9350, only one pod's logs were being dumped

# Testing steps

To validate these fixes we can create a new branch off of this one, modify a test to ensure it fails, open a PR so that CI runs, and observe that the generated artifact contains all expected output

This has not (yet) been done as the fixes are seemingly straightforward enough to be validated through static code analysis

# Checklist:
